org.uqbar.project.wollok/src/org/uqbar/project/wollok/scoping/WollokGlobalScopeProvider.xtend

```diff
@@ -114,12 +114,20 @@ class WollokGlobalScopeProvider extends DefaultGlobalScopeProvider {
 	 */
 	def loadResource(URI uri, ResourceSet resourceSet) {
 		try {
-			val resource = resourceSet.getResource(uri, true)
-			resource.load(#{})
-			resourceDescriptionManager.getResourceDescription(resource).exportedObjects
+			var Iterable<IEObjectDescription> exportedObjects
+			//checkResourceSet(resourceSet as XtextResourceSet)
+			exportedObjects = WollokResourceCache.getResource(uri)
+			if (exportedObjects == null) {
+				val resource = resourceSet.getResource(uri, true)
+				resource.load(#{})
+				exportedObjects = resourceDescriptionManager.getResourceDescription(resource).exportedObjects
+				WollokResourceCache.addResource(uri, exportedObjects)
+			}
+			exportedObjects
```

Comment:
And when is the cache invalidated or cleaned ?

I guess that for libraries (like SDK) that are read-only this is fine. But I wonder what happens when trying to load content from another file of your own project. Would it get out-dated ?